andela/Titan-Devs
# #163518708 Comment History

#### What does this PR do?
- Enable a user to edit the comment and track the comment history. 

#### Description of Task to be completed?
- Track the history of comment is it gets updated
- Enable fetching the history of the comment
-  Create edit comment route
- Enable fetching a particular comment
- Enable fetching comments related to an article. 
#### How should this be manually tested?
- Clone this repo 
- Check out the branch `ft-comment-history-163518708`
- Install all dependencies by running `yarn`
- Set the node environment to test `export NODE_ENV=test`
- Turn on your database server 
- Create a database by running `sequelize db:create`
- Run test by `yarn test`: You should see all tests passing
- Set the node environment to development `export NODE_ENV=devepment`
- Create the database by running `sequelize db:create`
- Run the migrations by `sequelize db:migrate`
- Start the server by running `yarn start`
- Use client tool like POSTMAN to perform activities below
- Create a user by consuming `POST: /api/v1/users`
- Log in to the application to get a token `POST: /api/v1/users/login` and grab the token
- Use the token in headers as you consume `POST: /api/v1/articles` to create article.
- Grab the article's slug and use and consume `POST: /api/v1/articles/:slug` to create a comment.
- Grab the comment Id and use to consume `PUT: /api/v1/articles/:slug/comments/:commentId` to edit the comment. You should see the comment history after editing the comment
- Use the commentId to consume `GET: /api/v1/articles/:slug/comments/:commentId/history` to see different versions of the comment and the time these versions were created. 

#### Any background context you want to provide?
The functionalities of fetching, updating, and deleting comment were not implemented before, so in this PR, these functionalities are included. Therefore, you can check `delete` and `get` on this route `/api/v1/articles/:slug/comments/:commentId` to see how it is implemented. 
#### What are the relevant pivotal tracker stories?
## 163518708
